R/epichains.R

```diff
@@ -55,19 +55,21 @@ new_epichains <- function(tree_df,
 #' @author James M. Azam
 #' @export
 epichains <- function(tree_df,
-                           index_cases,
-                           statistic,
-                           offspring_dist,
-                           stat_max,
-                           track_pop) {
+                      index_cases,
+                      statistic,
+                      offspring_dist,
+                      track_pop,
+                      stat_max = Inf) {
   # Check that inputs are well specified
-  checkmate::assert_data_frame(tree_df)
-  checkmate::assert_integerish(index_cases, null.ok = TRUE)
-  checkmate::assert_character(statistic, null.ok = TRUE)
+  checkmate::assert_data_frame(tree_df, min.cols = 3, min.rows = index_cases)
+  checkmate::assert_count(index_cases, positive = TRUE)
+  checkmate::assert_choice(statistic, choices = c("size", "length"))
   .check_offspring_func_valid(offspring_dist)
```

Comment:
This is not strictly part of this PR, but noticed it anyway. In {epidemics} I've tried to follow the {checkmate} naming style for input checking functions. So if this is an assert which errors when not true, could it be named `.assert_offspring_func_valid()`?

Reply:
Thanks. This was written a while back when I was not super familiar with the differences between assertions/checks/tests, so I'll create an issue to resolve this.